aiven/client/cli.py

```diff
@@ -1684,6 +1684,56 @@ def service__index_delete(self):
             index_name=self.args.index_name,
         )
 
+    @arg.project
+    @arg.service_name
+    @arg("--format", help="Format string for output, e.g. '{name} {retention_hours}'")
+    @arg.json
+    def service__m3__namespace__list(self):
+        """List M3 namespaces"""
+        namespaces = self.client.list_m3_namespaces(project=self.get_project(), service=self.args.name)
+        layout = [[
+            "name",
+            "type",
+            "resolution",
+            "options.retention_options.retention_period_duration",
+            "options.retention_options.blocksize_duration",
+        ]]
+        if not self.args.json:
+            # fix optional fields
+            for ns in namespaces:
+                ns['resolution'] = ns.get('resolution', '')
+                ns['options'] = ns.get('options', {})
+                ro = ns['options'].get('retention_options', {})
+                ns['options']['retention_options'] = ro
+                ro['retention_period_duration'] = ro.get('retention_period_duration', '')
+                ro['blocksize_duration'] = ro.get('blocksize_duration', '')
+        self.print_response(namespaces, format=self.args.format, json=self.args.json, table_layout=layout)
+
+    @arg.project
+    @arg.service_name
+    @arg.ns_name
+    def service__m3__namespace__delete(self):
+        """Delete M3 namespaces"""
+        self.client.delete_m3_namespace(project=self.get_project(), service=self.args.name, ns_name=self.args.ns_name)
+
+    @arg.project
+    @arg.service_name
+    @arg.ns_name
+    @arg.ns_type
+    @arg.ns_ret
+    @arg.ns_res
+    def service__m3__namespace__add(self):
+        """Add M3 namespaces"""
+        print(self.args)
```

Comment:
Maybe not print the arguments here. Or rather something bit prettier if we want to show output to the user.

Reply:
left my debug print there by mistake